### PR TITLE
[docker-orchagent][201911] Pass ASIC vendor information to swss docker as docker level environment variable

### PR DESCRIPTION
**Pull request by stephenxs**
#### Why I did it
Recently, the reserved buffer of admin-down ports is going to be reclaimed.
However, the way to do this differs among vendors.
We need to find a way to pass vendor information to swss docker.

Signed-off-by: Stephen Sun <stephens@nvidia.com>

#### How I did it
Fetch the ASIC vendor information when the docker is created and pass it to the docker as environment variable `ASIC_VENDOR`.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

File: files/build_templates/docker_image_ctl.j2 [hunk 1]
```diff
@@ -245,6 +245,11 @@ start() {
     echo "Creating new {{docker_container_name}}$DEV container with HWSKU $HWSKU"
     {%- endif %}
 
+    {%- if docker_container_name == "swss" %}
+    # Obtain the vendor name
+    ASIC_VENDOR=`$SONIC_CFGGEN -y /etc/sonic/sonic_version.yml -v asic_type`
```
Comment (neethajohn) [inline files/build_templates/docker_image_ctl.j2:250]: Use the 'sonic_asic_platform' var instead of defining  "ASIC_VENDOR"

Comment (stephenxs) [reply files/build_templates/docker_image_ctl.j2]: fixed.

**Comment by qiluo-msft** [comment]
You can get the info from inside swss container
```
sonic-cfggen -y /etc/sonic/sonic_version.yml -v asic_type
```

If you still want to add a PR, could you add the feature on master first and request a cherry-pick?

**Comment by stephenxs** [comment]
> You can get the info from inside swss container
> 
> ```
> sonic-cfggen -y /etc/sonic/sonic_version.yml -v asic_type
> ```
> 
> If you still want to add a PR, could you add the feature on master first and request a cherry-pick?

Thanks, Qi.
I think fetching it once when the container is created is better than fetching it every time the daemon starts in swss container.
In 202012 and above, we have similiar logic.

File: files/build_templates/docker_image_ctl.j2 (hunk 1 above)
Comment (qiluo-msft) [reply files/build_templates/docker_image_ctl.j2]: I agree with @neethajohn 's suggestion and it should go to master, 202012 and later braches.

**Comment by lguohan** [comment]
what is the motivation to move expose asic vendor to swss docker?

**Comment by neethajohn** [comment]
> what is the motivation to move expose asic vendor to swss docker?

Related to the comments in https://github.com/Azure/sonic-swss/pull/1837

File: files/build_templates/docker_image_ctl.j2 (hunk 1 above)
Comment (stephenxs) [reply files/build_templates/docker_image_ctl.j2]: @qiluo-msft PR to master created #8361

**Review by keboliu — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**